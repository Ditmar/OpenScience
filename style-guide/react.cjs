const { resolve } = require('node:path');
const project = resolve(process.cwd(), 'tsconfig.json');

module.exports = {
  env: {
    browser: true,
    es2021: true,
    es6: true,
    jest: true,
  },
  extends: [
    require.resolve('./rules/base'),
    require.resolve('./configurations/_base'),
    'plugin:react-hooks/recommended',
    'airbnb',
    'plugin:storybook/recommended',
    require.resolve('./configurations/typescript'),
    require.resolve('./configurations/vitest'),
  ],
  parserOptions: {
    project,
    babelOptions: {
      presets: ['@babel/preset-react'],
    },
  },
  plugins: ['react-refresh'],
  settings: {
    react: {
      version: 'detect',
    },
  },
  globals: {
    JSX: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    'react/button-has-type': 'off',
    'react-refresh/only-export-components': [
      'warn',
      { allowExportNames: ['meta', 'links', 'headers', 'loader', 'action'] },
    ],
  },
};
