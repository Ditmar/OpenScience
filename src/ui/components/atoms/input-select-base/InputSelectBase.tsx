import { useEffect, useState } from 'react';
import { Select } from '@mui/material';
import { selectSx } from './InputSelectBase.styles';
import type { IProps } from './types/IProps';

export function InputSelectBase(props: IProps) {
  const {
    error,
    disabled,
    borderRadius,
    size,
    children,
    colorVariant,
    valueSelect,
    onChange,
    menuProps,
    ...muiSelectProps
  } = props;
  const [keyboardFocus, setKeyboardFocus] = useState(false);

  useEffect(() => {
    const onKey = () => {
      setKeyboardFocus(true);
    };
    const onMouse = () => {
      setKeyboardFocus(false);
    };
    window.addEventListener('keydown', onKey);
    window.addEventListener('mousedown', onMouse);
    return () => {
      window.removeEventListener('keydown', onKey);
      window.removeEventListener('mousedown', onMouse);
    };
  }, []);

  const selectStyles = selectSx({ disabled, error, borderRadius, size, colorVariant });
  const focusCls = keyboardFocus ? 'keyboard-focus' : 'mouse-focus';

  return (
    <Select
      id="country-select"
      value={valueSelect}
      onChange={onChange}
      displayEmpty
      className={focusCls}
      disabled={disabled}
      inputProps={{ 'aria-label': 'Without label' }}
      sx={selectStyles}
      MenuProps={muiSelectProps.MenuProps}
    >
      {children}
    </Select>
  );
}
