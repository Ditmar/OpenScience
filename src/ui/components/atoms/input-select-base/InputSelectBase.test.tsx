import { render, screen } from '@testing-library/react';
import { InputSelectBase } from './InputSelectBase';
import type { ICountry } from '../country-list/interfaces/Country.interface';

export const options: ICountry[] = [
  { name: 'Argentina', code: 'ar', dialCode: '+51' },
  { name: 'Bolivia', code: 'bo', dialCode: '+51' },
  { name: 'Chile', code: 'cl', dialCode: '+51' },
  { name: 'PerÃº', code: 'pe', dialCode: '+51' },
];

describe('InputSelect Component', () => {
  it('renders disabled state', () => {
    render(<InputSelectBase disabled />);
    const select = screen.getByRole('combobox');
    expect(select).toHaveAttribute('aria-disabled', 'true');
  });

  it('matches snapshot in empty state', () => {
    const { container } = render(<InputSelectBase />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with error', () => {
    const { container } = render(<InputSelectBase error />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot disabled', () => {
    const { container } = render(<InputSelectBase disabled />);
    expect(container).toMatchSnapshot();
  });
});
