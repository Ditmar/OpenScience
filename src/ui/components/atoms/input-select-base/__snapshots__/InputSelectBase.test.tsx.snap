// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputSelect Component > matches snapshot disabled 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled mouse-focus MuiSelect-root css-1owj41t-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-disabled="true"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Without label"
      aria-labelledby="country-select"
      class="MuiSelect-select MuiSelect-outlined Mui-disabled MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="country-select"
      role="combobox"
    >
      <span
        aria-hidden="true"
        class="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
      disabled=""
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined Mui-disabled css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-w4cd9x"
      >
        <span
          aria-hidden="true"
          class="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputSelect Component > matches snapshot in empty state 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary mouse-focus MuiSelect-root css-qgf974-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Without label"
      aria-labelledby="country-select"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="country-select"
      role="combobox"
      tabindex="0"
    >
      <span
        aria-hidden="true"
        class="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-w4cd9x"
      >
        <span
          aria-hidden="true"
          class="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`InputSelect Component > matches snapshot with error 1`] = `
<div>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary mouse-focus MuiSelect-root css-cky9aa-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Without label"
      aria-labelledby="country-select"
      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="country-select"
      role="combobox"
      tabindex="0"
    >
      <span
        aria-hidden="true"
        class="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden="true"
      aria-invalid="false"
      class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
    >
      <legend
        class="css-w4cd9x"
      >
        <span
          aria-hidden="true"
          class="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
